src/App.tsx

```diff
@@ -9,47 +9,54 @@ import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
 
-export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+import { getUsers } from './api/users';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { setUsers } from './features/usersSlice';
+import { setItems, setLoaded, setError } from './features/postsSlice';
+import { setSelectedPost } from './features/selectedPostSlice';
 
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+export const App: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const posts = useAppSelector(state => state.posts.items);
+  const loaded = useAppSelector(state => state.posts.loaded);
+  const hasError = useAppSelector(state => state.posts.hasError);
+  const selectedPost = useAppSelector(state => state.selectedPost);
+  const author = useAppSelector(state => state.author);
 
   function loadUserPosts(userId: number) {
-    setLoaded(false);
+    dispatch(setLoaded(false));
 
     getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
+      .then(data => dispatch(setItems(data)))
+      .catch(() => dispatch(setError(true)))
+      .finally(() => dispatch(setLoaded(true)));
   }
 
   useEffect(() => {
     // we clear the post when an author is changed
     // not to confuse the user
-    setSelectedPost(null);
+    dispatch(setSelectedPost(null));
 
     if (author) {
       loadUserPosts(author.id);
     } else {
-      setPosts([]);
+      dispatch(setItems([]));
```

Comment:
Fix it 
<img width="1222" alt="image" src="https://github.com/mate-academy/react_redux-list-of-posts/assets/64750958/02efaf87-78f9-4f2b-ad20-1cd3cedd6b68">